VisualPinball.Unity/VisualPinball.Unity/VPT/PhysicsMaterialComponent.cs

```diff
@@ -1,38 +1,80 @@
-﻿// Visual Pinball Engine
-// Copyright (C) 2022 freezy and VPE Team
-//
-// This program is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-//
-// This program is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with this program. If not, see <https://www.gnu.org/licenses/>.
-
-// ReSharper disable InconsistentNaming
-
-using UnityEngine;
-
-namespace VisualPinball.Unity
-{
-	/// <summary>
-	/// A physical material used by the physics engine<p/>
-	///
-	/// Materials are actually a big deal in VP, authors as well as players
-	/// tweak them all the time, so getting those from external assets instead
-	/// of writing them into the scene seems a good plan.
-	/// </summary>
-	[CreateAssetMenu(fileName = "PhysicsMaterial", menuName = "Visual Pinball/Physics Material", order = 100)]
-	public class PhysicsMaterial : ScriptableObject
-	{
-		public float Elasticity;
-		public float ElasticityFalloff;
-		public float Friction;
-		public float ScatterAngle;
-	}
-}
+﻿// Visual Pinball Engine
+// Copyright (C) 2022 freezy and VPE Team
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// ReSharper disable InconsistentNaming
+
+using UnityEngine;
+using UnityEditor;
+using Unity.Collections;
+
+namespace VisualPinball.Unity
+{
+	/// <summary>
+	/// A physical material used by the physics engine<p/>
+	///
+	/// Materials are actually a big deal in VP, authors as well as players
+	/// tweak them all the time, so getting those from external assets instead
+	/// of writing them into the scene seems a good plan.
+	/// </summary>
+	[CreateAssetMenu(fileName = "PhysicsMaterial", menuName = "Visual Pinball/Physics Material", order = 100)]
+	//[CustomEditor(typeof(PhysicsMaterial))]
+	public class PhysicsMaterialComponent : ScriptableObject
```

Comment:
Any particular reason to rename this? It's not a component, so it's a bit confusing. If you wanna be more precise, I'd propose `PhysicsMaterialAsset`, which is how scriptable objects are stored.

Reply:
*Asset would be good. My problem was that we had a struct and the asset, named the same. Renaming will break the current table again though, but can do this tomorrow.